### ISSUE TITLE
Histogram function doesn't fail for non-bool values. 

### ISSUE THREAD

**Issue by dubey-sukhsagar** [quantumlib/Cirq#5926]
**The histogram function has a default value for the fold_func parameter that calculates the integer value for the binary big endian number got from measurement results assuming qubits, not qudits**

**Take any qudit circuit and you will find below result is wrong (as per binary) 
samples = sim.run(circuit, repetitions=10)
result = samples.histogram(key='out')**

```python
# Create a circuit with three qutrit gates.
>>> q0 = cirq.LineQid(0,dimension=3)
>>> q1 = cirq.LineQid(1, dimension=3)
>>> q2 = cirq.LineQid(2, dimension=3)
>>> circuit = cirq.Circuit([
    QutritPlusGate()(q0), #qutrit X gate
    QutritPlusGate()(q0),
    QutritPlusGate()(q1),
    QutritPlusGate()(q1),
    QutritPlusGate()(q2),
    QutritPlusGate()(q2),
    cirq.measure(q0,q1,q2, key="out")
])
>>> samples = sim.run(circuit, repetitions=1)
>>> res = samples.histogram(key='out')
>>> print(circuit)
>>> print(samples)
>>> print(res)

```

For above circuit res should be 26 but I am getting 7.
![image](https://user-images.githubusercontent.com/26180953/196505964-69089b98-c6de-43cb-9cf4-1f19a4b2e9aa.png)

**Cirq version**
0.15.0

**Comment by seunomonije** [comment]
From Cirq Sync: need to investigate more. Histogram fn clearly is not working properly for qutrits. 

Tanuj: There might be a workaround if you customize the fold function in the histogram creator. Right now it is default for base 2. Potential change to make the default for any base?

**Comment by Su760** [comment]
Hi! I'm a student in the CodePath AI301 program and I'd like to work on this issue. I'll start by reproducing the problem locally this week. Any pointers on where to start in the codebase would be appreciated!

**Comment by mhucka** [comment]
> Hi! I'm a student in the CodePath AI301 program and I'd like to work on this issue. I'll start by reproducing the problem locally this week. Any pointers on where to start in the codebase would be appreciated!

@Su760 Thank you for your interest! Before you start, please make sure to read the contribution guideliens in CONTRIBUTING.md. Note especially that contributors need to be able to [sign the CLA](https://github.com/quantumlib/Cirq/blob/main/CONTRIBUTING.md#contributor-license-agreement), and the terms of the CLA are that contributions must be your original creation. Using assistance is one thing, but submitting code produced entirely by AI tools does not qualify as a person's original creation.

**Comment by Su760** [comment]
Hi! I've submitted a PR to address this issue: #8138. It adds an optional qid_shape parameter to histogram() so qudit circuits use the correct base. Happy to get feedback!

**Comment by pavoljuhas** [comment]
> Hi! I'm a student in the CodePath AI301 program and I'd like to work on this issue.

Just curious - is this a class assignment?

**Comment by Su760** [comment]
Yes. I'm taking the CodePath AI301 open source program course this summer, where the goal is to make an open source contribution.
